labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/impl/KeyValueStorageEncryptedSharedPreferences.kt

```diff
@@ -126,14 +140,19 @@ public class KeyValueStorageEncryptedSharedPreferences constructor(
     }
 
     private fun createOrGetMasterKey(): MasterKey {
-        // Note setRequestStrongBoxBacked is set to false for speed reasons for now.
-        // Will be configurable in the future.
         return MasterKey.Builder(appContext, STORAGE_MASTER_KEY_ALIAS)
             .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
-            .setRequestStrongBoxBacked(false)
+            .setRequestStrongBoxBacked(getFromPreferredType())
             .build()
     }
 
+    private fun getFromPreferredType(): Boolean {
```

Comment:
The name could be better. Doesn't say what we get from the preferred type here

Reply:
Agreed, good point